src/frontend/tests/core/features/actionsMainPage-shard-1.spec.ts

```diff
@@ -128,7 +139,9 @@ test("search components", async ({ page }) => {
 
   while (modalCount === 0) {
     await page.getByText("New Flow", { exact: true }).click();
-    await page.waitForTimeout(3000);
+    await page.waitForSelector('[data-testid="modal-title"]', {
+      timeout: 30000,
```

Comment:
i think we can reduce the timeout to 3000 since it just waited for that timeout before